### DISCUSSION TITLE
Select environment configuration for Angular w/ firebase hosting web frameworks

### DISCUSSION THREAD

**Discussion by kanehjeong** [firebase/firebase-tools#6613] [Q&A]
Hello all,

I am using the firebase hosting web frameworks experimental feature with an Angular ssr project. I've got everything set up correctly and everything seems to deploy fine, _except_ I started getting a bunch of CORs errors because my staging url was not whitelisted. Then I realized that it builds and deploys using the production build pipeline by default. I couldn't find anywhere in the documentation that allows you to select an environment configuration (by default if triggering a build manually, you can do `ng build --configuration staging`.

TLDR running `firebase deploy` will build my Angular's project w/ the production configuration only with seemingly no way to change it. Is there a way to specify a configuration to use?

**Comment by jamesdaniels** [comment]
`FIREBASE_FRAMEWORKS_BUILD_TARGET=staging firebase deploy` should achieve the desired outcome

**Reply by DaveHowie** [reply]
Sorry, I realize this is old, but I'm running into the same issue as that reported by kanehjeong...

I have Firebase App Hosting setup with CI/CD on push to my GitHub repo. There is no viewable/editable Cloud Build trigger in GCP. The Cloud Build steps are all "under the covers", triggered on push to my GitHub repo.

With that being said, I'm at a loss as to where in the Cloud Build process I would specify: FIREBASE_FRAMEWORKS_BUILD_TARGET=staging

**Reply by DaveHowie** [reply]
Never mind...

I solved my problem simply by changing the scripts section of my package.json file as follows:

Changed this...


    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "watch": "ng build --watch",
    "test": "ng test"


...to this...

    "ng": "ng",
    "start": "ng serve --configuration $ENVIRONMENT_NAME",
    "build": "ng build --configuration $ENVIRONMENT_NAME",
    "watch": "ng build --watch --configuration $ENVIRONMENT_NAME",
    "test": "ng test"

Note: If you are using Cloud Build packs to build your Angular app for hosting in FireBase App Hosting, as I am doing, and if you look close enough at the build logs, you may see the following warning:

`19:23:28 | WARNING: *** You are using a custom build command (your build command is NOT 'ng build'), we will accept it as is but will error if output structure is not as expected ***`

This is a result of including the command line arguments for the start, build and watch scripts, but it does not cause any issues (hence why it is a warning); I am getting the expected configuration as named by the ENVIRONMENT_NAME variable.
